scripts/dcp1-version/populate_dcp_version.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import os
+import uuid
+from datetime import datetime
+
+from bson import JAVA_LEGACY, CodecOptions
+from pymongo import MongoClient
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+DCP1_GS_FILES_LS = os.environ.get('DCP1_GS_FILES_LS')
+DB_HOST = os.environ.get('DB_HOST', 'localhost')
+DB_PORT = os.environ.get('DB_PORT', 27017)
+
+DB_CLIENT = MongoClient(DB_HOST, 27017)
+DB = DB_CLIENT.get_database('admin')
+
+DCP_VERSION_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+COLLECTION_MAP = {
+    'project': 'project',
+    'cell_line': 'biomaterial',
+    'cell_suspension': 'biomaterial',
+    'collection_protocol': 'protocol',
+    'differentiation_protocol': 'protocol',
+    'dissociation_protocol': 'protocol',
+    'donor_organism': 'biomaterial',
+    'enrichment_protocol': 'protocol',
+    'ipsc_induction_protocol': 'protocol',
+    'library_preparation_protocol': 'protocol',
+    'organoid': 'biomaterial',
+    'process': 'process',
+    'sequence_file': 'file',
+    'sequencing_protocol': 'protocol',
+    'specimen_from_organism': 'biomaterial',
+    'supplementary_file': 'file'
+}
+
+
+def load_lines_from_file(file: str):
+    with open(file) as f:
+        data = [line.rstrip() for line in f]
+        return data
+
+
+def find_type_uuid_version(line: str):
+    filename = line.split('/')[-1]
+    type = line.split('/')[-2]
+    parts = filename.split('_')
+    uuid_str = parts[0]
+    version = parts[1].replace('.json', '')
+    return type, uuid_str, version
+
+
+def find_doc_by_uuid(collection: str, uuid_str: str):
+    LOGGER.info(f'finding {uuid_str} in {collection}')
+    # need to use java legacy uuid representation
+    # https://stackoverflow.com/questions/26712600/mongo-uuid-python-vs-java-format/31061472
+    db_collection = DB.get_collection(collection, CodecOptions(uuid_representation=JAVA_LEGACY))
+
+    doc = db_collection.find_one(
+        {
+            'uuid.uuid': uuid.UUID(uuid_str),
+            'isUpdate': False
+        }
+    )
+
+    if not doc:
+        LOGGER.error(f'No doc found for {uuid_str}')
+
+    return doc
+
+
+def determine_updates(doc: dict, date_obj: datetime):
```

Comment:
This function checks if an update is needed and also generates some data. The function that uses it calls is, update_dcp_versions, checks for its return value, and if not null, updates the database.
I find this a bit convoluted.

Have you considered the following flow?

1. check if update is necessary
2. if so build the `update` object
3. update the db

Reply:
sounds good 👍, I applied it in my latest commit.

Reply:
Thanks.
Have you considered renaming the  `determine_updates` function to `is_update_necessary`?

Reply:
The method is returning the patch dictionary.